django/db/migrations/recorder.py

```diff
@@ -18,19 +19,26 @@ class MigrationRecorder:
     If a migration is unapplied its row is removed from the table. Having
     a row in the table always means a migration is applied.
     """
-
-    class Migration(models.Model):
-        app = models.CharField(max_length=255)
-        name = models.CharField(max_length=255)
-        applied = models.DateTimeField(default=now)
-
-        class Meta:
-            apps = Apps()
-            app_label = "migrations"
-            db_table = "django_migrations"
-
-        def __str__(self):
-            return "Migration %s for %s" % (self.name, self.app)
+    _migration_class = None
+
+    @classproperty
+    def Migration(cls):
+        if cls._migration_class is None:
+            class Migration(models.Model):
+                app = models.CharField(max_length=255)
+                name = models.CharField(max_length=255)
+                applied = models.DateTimeField(default=now)
+
+                class Meta:
+                    apps = Apps()
+                    app_label = "migrations"
```

Comment:
While we're modfying these lines, you could switch all these strings to use single quotes. And perhaps the property deserves a docstring or comment explaining the "why" of the classproperty approach.